### PR TITLE
Add schema helper methods, modify device level tests, and fix some docs

### PR DESCRIPTION
**Pull request by ajberdy**
*Issue #, if available:*

*Description of changes:*

Reverting the partial fix of adding null to certain Optional types until a better long-term solution is found. Changed a typo in documentation. Added helper methods for the base schema and header class.

*Testing done:*

Tox

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [X] I have read the [CONTRIBUTING](https://github.com/aws/amazon-braket-schemas-python/blob/main/CONTRIBUTING.md) doc
- [X] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/amazon-braket-schemas-python/blob/main/CONTRIBUTING.md#commit-your-change)
- [X] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-braket-schemas-python/blob/main/README.md) and [API docs](https://github.com/aws/amazon-braket-schemas-python/blob/main/CONTRIBUTING.md#documentation-guidelines) (if appropriate)

#### Tests

- [X] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [X] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/unit_tests/braket/schema_common/test_schema_base.py [hunk 1]
```diff
@@ -62,3 +62,14 @@ def test_parse_raw_schema():
     )
     assert schema == BraketSchemaBase.parse_raw_schema(schema.json())
     assert isinstance(schema, TaskMetadata)
+
+
+def test_get_schema_class():
```
Comment (jheckey) [inline test/unit_tests/braket/schema_common/test_schema_base.py:67]: Add a test for the negative case as well - make sure an undefined module will throw an error.

**Review by jheckey — DISMISSED** [review]

**Review by jheckey — APPROVED** [review]
